api/openapi-spec/conventions-server..yaml

```diff
@@ -0,0 +1,133 @@
+openapi: 3.0.0
+info:
+  title: Conventions server 
+  description: >-
+    "a sample conventions server"
+  version: 1.0.0
+  license:
+    name: Apache-2.0
+    url: 'https://www.apache.org/licenses/LICENSE-2.0.html'
+paths:
+  /webhook:
+    post:
+      description: "a sample conventions server"
+      responses:
+        200:
+          description: Successfully applied defined conventions
+          content:
+            "application/json":
+              schema:
+                $ref: '#/components/schemas/PodConventionContext'
+        400:
+          description: Error decoding PodConventionContext from the request body or request body is nil 
+        500:
+          description: Internal server error
+          content:
+            "application/json":
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+components:
+  schemas:
+    PodConventionContext: 
+      type: object
+      properties:
+        apiVersions: 
+          type: string
+        kind: 
+          type: string
+        metadata:
+          type: array 
+          items: 
+            type: string
+        spec:
+          $ref: "#/components/schemas/PodConventionContextSpec"
+        status:
+          $ref: "#/components/schemas/PodConventionContextStatus"
+    PodConventionContextSpec:
+      type: object
+      properties:
+        template:
+          $ref: "#/components/schemas/PodTemplateSpec"
+        imageConfig:
+          type: array 
+          description: an array of imageConfig objects
+          items:
+            $ref: "#/components/schemas/ImageConfig"
+    PodTemplateSpec:
+      type: object 
+      properties:
+        spec: 
+          type: object
```

Comment:
there are a *lot* of available fields that go quite deep. Similar to the discussion about the ggcr config, we should consider leaving it open ended as to not artificially restrict new fields from being added in later versions of k8s, and not exposed by a convention server. This relates to #60